### PR TITLE
[refine](column) refine append_data_by_selector code 

### PR DESCRIPTION
**Pull request by Mryange**
### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [x] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run buildall

**Comment by Mryange** [comment]
run p0

**Comment by Mryange** [comment]
run buildall

File: be/src/vec/columns/column.h [hunk 1]
```diff
@@ -503,13 +503,17 @@ class IColumn : public COW<IColumn> {
...
       * Selector must contain values from 0 to num_columns - 1.
-      * For default implementation, see scatter_impl.
+      * For default implementation, see column_impl.h
       */
     using ColumnIndex = UInt64;
     using Selector = PaddedPODArray<ColumnIndex>;
 
+    // The append_data_by_selector function requires the column to implement the insert_from function.
+    // In fact, this function is just calling insert_from but without the overhead of a virtual function.
+
```
Comment (HappenLee) [inline be/src/vec/columns/column.h:513]: remove  the useless space line

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**